### ISSUE TITLE
Epic mixup of files after disk restoration

### ISSUE THREAD

**Issue by lenlo** [syncthing/syncthing#5831]
The main SSD of Mega, my Mac mini server, failed on me a few weeks ago. Fortunately, it was under warranty, so I recently received a replacement and restored the contents from backup. After the backup was complete and the server was rebooted, Syncthing started up on Mega and quickly ran out of open files. Strange, I thought, it had never done that before. Still, I just up'ed the max number of open files from 256 to 1024 and let it continue.

The next day, I picked up Neo, my MacBook Pro laptop, and noticed that it was out of disk space. Huh, what had happened? A closer inspection showed that every folder that was being synced with Mega was full of files -- in fact *every* synced folders seemed to have a copy of *all* the files from *all* the folders that Mega was syncing with *any* other host!

That is, if Mega had folders A, B, and C, and Neo was syncing A and B with Mega, Neo's A and B folders now *both* appeared to contain *all* the files from A, B, and C all mixed together. The same was true for the A folders on *all* the other hosts that was syncing with Mega and/or Neo.

Not only has this created a mess of epic proportions, it's also a giant security hole since the private files in B now have been distributed to a whole slew of public hosts that only were supposed to have the A files.

And as if that wasn't bad enough, when I frantically started deleting the misplaced files and folders, *Syncthing started putting them back again!*

That is, when I started deleting the B files from the A folder, Syncthing promptly recreated them again! At first I thought I had made a mistake when deleting them, but when I deleted them again and listed the files in the directory, they were clearly gone -- only to reappear a few seconds later.

To be fair, it seems to mostly(? only?) be the subdirectories in which the files reside, not necessarily the files themselves -- I wasn't paying full attention when it first happened, and now when I'm trying to reproduce it, I'm only seeing (empty) subdirectories being recreated. Looking at syncthing.log, I see a lot of:

[SNPSJ] 12:24:12 INFO: Puller (folder A, item "bsubdir/bfile"): no connected device has the required version of this file

This seem to cause the recreation of bsubdir.

The only solution I've been able to come up with is to delete all Syncthing configurations on all hosts and recreate them from scratch as just rescanning the folders seems to make no change. I've put aside a copy of the config files on Mega together with syncthing.log. I had a quick look at syncthing.log from after the restoration, but I couldn't find anything obvious in it except for a whole bunch of "too many open files" errors. I can make them available if it would help, but I would first need to anonymize the contents.

Syncthing Version: v1.1.4
OS Version: macOS 10.14.5 / Ubuntu 16.04.5 LTS & 18.04.2 LTS

**Comment by lenlo** [comment]
Quick update: Apart from the empty subdirs that Syncthing is insisting on recreating after I delete them, I'm also seeing a fair number of .syncthing.xxx.tmp files in there.

**Comment by AudriusButkevicius** [comment]
Support happens on the forums.

**Comment by lenlo** [comment]
Huh? I'm not looking for support, I wanted to file a bug where Syncthing managed to get into an extremely twisted state simply because a disk containing its configuration and a good chunk of its data files was restored from backup. I have no reason to believe that the backed up files were corrupted in any way.

**Comment by lenlo** [comment]
Are you implying that this is expected behavior?

**Comment by imsodin** [comment]
It's not clear what exactly happened, so this is a support request until the situation is cleared up and a problem identified.

**Comment by lenlo** [comment]
I thought I already told you what happened. An SSD hosting Syncthing's database and some of corpus had a catastrophic failure on my server. When the backup was restored to a new, identical SSD, Syncthing started "sharing" the files between *all* the different synced folders. There were no known errors with the backup, nor with the restoration process, so I have no reason to believe that Syncthing's state got corrupted in the process.

Clearly, it wasn't able to function properly after it was restored to the new disk, but why I have no idea. Perhaps it just wasn't able to continue with the state as it was saved or maybe the disk UUID change confused it, I have no idea. Either way, it suggests fragility that IMHO ought to be looked at. But if you have no interest in this, I will delete my saved state and move on. Preferably to a different syncing solution since I find this behavior extremely unsettling.

**Comment by AudriusButkevicius** [comment]
So issues github are things that need to be fixed and its clear what it is. Can you quantify what exactly needs to be fixed, can you provide exact steps that produce this issue (that you yourself verified that we can independently of your state reproduce this so we could look into and understand it)?


Sure you had some random state of affairs and bad things happened, but it doesn't explain what exactly we need to fix, nor what caused it, hence why it should be a discussion in the forums to iron this out and not a "shout in the forest" ticket with nothing actionable.

Move to a different solution for all that I care, but I will not leave issues lying around that do not have a sensible definition of what needs to be fixed around.

**Comment by imsodin** [comment]
Let me try again: Neither Audrius nor me said that what you describe sounds like a good thing to happen so case closed. We asked you to please take it to the forum, where in addition to contributors there's a whole community with experience in various use-cases on a diversity of system that may identify what went wrong in your case. If along this way an actual actionable issue with Syncthing is identified, a Github issue will be opened and hopefully a solution found soon. In the meantime, I suggest you value the advice of those people that will probably solve your problem and head to the forum.

**Comment by lenlo** [comment]
I very much appreciate your effort in making Syncthing a great and well working file syncing system, but if all you care about are easily reproducible problems with well defined fixes, then it sounds like we may have very different standards of what constitutes a bug and what is worth documenting in a bug tracking system. Be that as it may, I acknowledge that I'm a freeloading user in this case and not on the other side of the fence. My bug submission is yours to do with as you please.

Once again, I wasn't looking for help in fixing my setup, I wanted to report a behavior with Syncthing that seemed exceedingly out of line which I thought you might find of interest. As for "a random state of affairs," the only random part I can think of is the state that Syncthing's database was in at the time of the backup, something that I would have expected it to be able to recover from without melting down in the way it did. After all, process deaths and kernel panics can happen just as easily too.

I have already rebuilt my environment back to a fully operational state, so I don't need any help with that. If you nevertheless think that posting a report of my experience to the support forum would be helpful for the future quality of the product, then sure, I can do that. I don't see the point myself, but maybe I'm just not understanding how your process is meant to work.

**Comment by canton7** [comment]
@lenlo There will doubtless need to be some back-and-forth to fill in the (inevitable) gaps in your original description, while the community tries to understand the root cause of your issue.

This project's policy (for better or for worse) is that that process happens on the forums. GitHub issues are used once the cause of the problem has been identified. 

So yes, please do move your post to the forums 👍
